Observable/Classes/Base.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+public class BaseObservable<T> {
+    
+    public typealias Observer = (T, T?) -> Void
+    
+    private var observers: [Int: (Observer, DispatchQueue?)] = [:]
+    private var uniqueID = (0...).makeIterator()
+    
+    let lock: Lock = Mutex()
+    var _value: T? {
+        didSet {
+            let val = _value!
```

Comment:
If the _value becomes nil, which is a valid option, this will crash

Reply:
Set `nil` value is internal property, only inside framework you can set nil value. Outside it's not available.

Reply:
I have tested it, and you are right.